lib/AppInfo/Application.php

```diff
@@ -93,4 +95,30 @@ public function registerHooks() {
 			'loginName2UserName'
 		);
 	}
+
+	public function registerEventListener() {
+		$container = $this->getContainer();
+		$server = $container->getServer();
+
+		$uProxy = $container->query(User_Proxy::class);
+		$eventDispatcher = $server->getEventDispatcher();
+		$eventDispatcher->addListener(UserExtendedAttributesEvent::USER_EXTENDED_ATTRIBUTES, function (UserExtendedAttributesEvent $event) use ($uProxy) {
+			$targetUser = $event->getUser();
+			if ($targetUser->getBackendClassName() !== 'LDAP') {
+				// If the user doesn't come from LDAP, there is nothing to do here.
+				return;
+			}
+
+			try {
+				$attrs = $uProxy->getExposedAttributes($targetUser->getUID());
+
+				$event->setAttributes('user_ldap_state', 'OK');
+				foreach ($attrs as $key => $value) {
+					$event->setAttributes("user_ldap_attr_{$key}", $value);
```

Comment:
This method is called `setAttributes` but actually it only sets one attribute.
But I see that `UserExtendedAttributesEvent` comes from core, so we can't adjust the method name now!

Reply:
Yeah, I don't think we can do much to fix this. We might include another method in the event, but we'll have to check the ownCloud version here.